administrator/components/com_languages/helpers/multilangstatus.php

```
@@ -81,17 +71,31 @@ public static function getContentlangs()
 	 */
 	public static function getSitelangs()
 	{
-		// Check for published Site Languages.
-		$db = JFactory::getDbo();
-		$query = $db->getQuery(true)
-			->select('a.element AS element')
-			->from('#__extensions AS a')
-			->where('a.type = ' . $db->quote('language'))
-			->where('a.client_id = 0')
-			->where('a.enabled = 1');
-		$db->setQuery($query);
-
-		return $db->loadObjectList('element');
+		static $sitelangs = null;
+
+		if (empty($sitelangs))
+		{
+			// Check for published Site Languages.
+			$db = JFactory::getDbo();
+			$query = $db->getQuery(true)
+				->select('a.element AS element')
+				->from('#__extensions AS a')
+				->where('a.type = ' . $db->quote('language'))
+				->where('a.client_id = 0')
+				->where('a.enabled = 1');
+			$db->setQuery($query);
+
+			try
+			{
+				$sitelangs = $db->loadObjectList('element');
+			}
+			catch (RuntimeException $e)
+			{
+				throw new Exception($e->getMessage(), 500);
```

Is there any reason why we have to catch these exceptions and rethrow them and not just let them bubble up?